framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/AbpDbContextOptions.cs

```diff
@@ -7,19 +7,22 @@ namespace Volo.Abp.EntityFrameworkCore
 {
     public class AbpDbContextOptions
     {
-        internal List<Action<AbpDbContextConfigurationContext>> DefaultPreConfigureActions { get; set; }
+        internal List<Action<AbpDbContextConfigurationContext>> DefaultPreConfigureActions { get; }
 
         internal Action<AbpDbContextConfigurationContext> DefaultConfigureAction { get; set; }
 
-        internal Dictionary<Type, List<object>> PreConfigureActions { get; set; }
+        internal Dictionary<Type, List<object>> PreConfigureActions { get; }
 
-        internal Dictionary<Type, object> ConfigureActions { get; set; }
+        internal Dictionary<Type, object> ConfigureActions { get; }
+        
+        internal Dictionary<Type, Type> DbContextReplacements { get; }
```

Comment:
How about `Dictionary<Type, List<Type>> DbContextReplacements { get; }`

**key** is **DbContext type** that replace the types of **value**. 

```cs
services.Configure<AbpDbContextOptions>(opts =>
{
    opts.DbContextReplacements.GetOrAdd(typeof(TDbContext), () => new List<Type>())
        .AddIfNotContains(dbContextType);
});
```

```cs
internal Type GetReplacedTypeOrSelf(Type dbContextType)
{
    return DbContextReplacements.Where(x => x.Value.Contains(dbContextType)).Select(x => x.Key).FirstOrDefault()
           ?? dbContextType;
}
```

Reply:
`Dictionary<Type, Type>` is faster than sequential search. What is the benefit of that approach?

Reply:
For me, it is easier to understand. And no need for `while (true)`

![image](https://user-images.githubusercontent.com/6908465/108045536-89802900-707e-11eb-84e5-4a12f3b93438.png)

Reply:
While-true is also needed for your case. It is actually an edge case. If A is replaced by B, then B is replaced then C, it finds C for A. It could be simplified, however using a `Dictionary<Type, Type>` is faster and pretty simple.

Reply:
If nothing else, can you please merge, I am working another feature based on it :)